cmd/init.js

```diff
@@ -39,12 +39,22 @@ module.exports = (ipc) => async function init (cmd) {
   try {
     await output(false, await require('../init')(link, dir, { ipc, autosubmit: yes, force, defaults, header }))
   } catch (err) {
-    if (err.code !== 'ERR_PERMISSION_REQUIRED') throw err
-    const explain = 'Be sure that software is trusted before using it\n' +
-    '\nType "TRUST" to allow template initialization or anything else to exit\n\n'
-    const ask = 'Trust template'
-    const act = 'Use pear init again to initalize from trusted template'
-    await trust({ ipc, key: err.info.key, message: err.message, explain, ask, act })
+    if (err.code !== 'ERR_PERMISSION_REQUIRED') {
+      throw err
+    } else {
+      if (err.info && err.info.encrypted) {
+        const explain = 'This template is encrypted.\n' +
+          '\nEnter the password to init from the template.\n\n'
+        const message = 'Added encryption key, run init again to complete it.'
+        await password({ ipc, key: err.info.key, explain, message })
+      } else {
+        const explain = 'Be sure that software is trusted before using it\n' +
+          '\nType "TRUST" to allow template initialization or anything else to exit\n\n'
+        const ask = 'Trust template'
+        const act = 'Use pear init again to initalize from trusted template'
+        await trust({ ipc, key: err.info.key, message: err.message, explain, ask, act })
+      }
+    }
```

Comment:
this is what iface `permit` should actually do 

this code from 45-57 here should just be one line

```
await permit({ ipc, info: err.info, message: err.message })
```

and then permit should contain the conditional logic to then call `trust` / `password `

as for the current iface `permit`, it's just inline its current function body into `trust` and `password`

Reply:
the issue is that `trust` and `password` all show different messages for stage, dump, run, init and info. Thats why we pass `explain`, `ask` and `act` as arguments.

Reply:
@rafapaezbas right right- so permit({ dialog: 'stage', .... })

and then the permit function owns all the output strings, so it passes the correct explain, ask, act args to trust and password

Reply:
done

Reply:
@rafapaezbas 
> this code from 45-57 here should just be one line

you're still not calling permit directly?

Reply:
done